### PR TITLE
Add Week3 Homework - Oguzhan

### PR DESCRIPTION
**Pull request by oguzhanturgut**
- Add async-await and try-catch, use fetch
- Add OOP version

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+  function fetchJSON(url) {
+    return fetch(url).then(response => {
+      if (!response.ok) {
+        throw new Error(`Error occured ${response.status} ${response.statusText}`);
```
Comment (remarcmij) [inline homework/index.js]: Make sure to correct spelling errors flagged by VSCode: `occured` -> `occurred`.

File: homework/index.js [hunk 2]
```diff
@@ -30,17 +24,156 @@
...
+      id: 'wrapper',
     });
+    createAndAppend('select', header, {
+      id: 'select',
+      'aria-label': 'HYF Repositories',
+    });
+
+    try {
+      const repositories = await fetchJSON(url);
+      const repository = initializePage(repositories);
```
Comment (remarcmij) [inline homework/index.js]: It is not immediately obvious what `initializePage` returns or should return. The function name gives no clue. I had to inspect the code to discover that it effectively returns the first repo from the list. Therefore, I would prefer this explicit code:

```js
try {
  const repositories = await fetchJSON(url);
  initializePage(repositories);
  fetchAndRender(repositories[0]);
} catch (error) {
  renderError(error);
}
```

File: homework-classes/Repository.js [hunk 3]
```diff
@@ -13,8 +13,47 @@ class Repository {
...
+
+  /**
+   * Returns a row for repository container with desired label and content
+   *
+   * @param {HTMLElement} parent Parent tbody element
+   * @param {string} label Label text
+   * @param {string} content Content text
+   * @returns Repo info row
+   */
+  addRepoInfoRow(parent, label, content) {
```
Comment (remarcmij) [inline homework-classes/Repository.js]: It seems you have grasped the concept of JSDoc, as applied to your own method.

File: homework/index.js (hunk 1 above)
Comment (oguzhanturgut) [reply homework/index.js]: Fixed typo. Thanks for the tip

File: homework/index.js (hunk 2 above)
Comment (oguzhanturgut) [reply homework/index.js]: Changed following your advice

File: homework-classes/Repository.js (hunk 3 above)
Comment (oguzhanturgut) [reply homework-classes/Repository.js]: Thanks. It is a very handy tool for generating documentation.

**Comment by oguzhanturgut** [comment]
Thanks for your kind and patient manner on teaching such challenging topics in JS. I will miss your lecture summaries after the class.
That was a pleasure for me attending your lectures
See you soon 👋

**Comment by remarcmij** [comment]
Thank you for expressing your appreciation. Good luck with the test and with the rest of the curriculum. See you again soon.

**Pull request closed without merging**